shuup/simple_cms/admin_module/views.py

```diff
@@ -128,6 +113,9 @@ def clean_parent(self):
     def save(self, commit=True):
         if not hasattr(self.instance, "shop") or not self.instance.shop:
             self.instance.shop = get_shop(self.request)
+        if self.cleaned_data.get("page_type") == PageType.REVISIONED:
```

Comment:
Could this be done in model level instead since likely not something the admin module need to know about?

Reply:
I thought it first, but I didn't want to introduce `def save` into the `Page` Model. Now that I think of it, it might make sense as it dries out the code quite a bit. Thanks